### PR TITLE
fix(response): use never return type for abort method

### PR DESCRIPTION
**Pull request by targos**
I tried to use the new feature that TypeScript added for assertions functions to fix the type of `abortIf` but couldn't find a way. `abortIf` is implemented in the opposite way (throws if the condition is truthy) than usual assertions.

Ref: https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#assertion-functions

### CODE REVIEW

**Comment by targos** [comment]
Ping. Even if we don't find a solution for `abortIf`, I think this change is important to get in.

**Comment by thetutlage** [comment]
Yeah, we need inverse of `asserts <val>`. Maybe we can also add another function `abortUnless`?

**Comment by targos** [comment]
> Yeah, we need inverse of `asserts <val>`. Maybe we can also add another function `abortUnless`?

Do you want me to add that to this PR?

**Comment by thetutlage** [comment]
Yeah, let's get them out together

**Comment by targos** [comment]
Done.

File: test/response.spec.ts [hunk 1]
```diff
@@ -1066,4 +1066,37 @@ test.group('Response', (group) => {
 
     await supertest(server).get('/').expect(200)
   })
+
+  test('abortUnless: abort request when condition is falsy', async (assert) => {
+    const server = createServer((req, res) => {
+      const config = fakeConfig()
+      const response: Response = new Response(req, res, config)
```
Comment (targos) [inline test/response.spec.ts]: See https://github.com/microsoft/TypeScript/pull/33622

It is necessary here to explicitly declare the type of `Response`. In practice in Adonis, the type is already defined in the context so the user doesn't have to do anything.

Example with the change, where TS correctly determined that the code after the call is unreachable:
image

**Comment by thetutlage** [comment]
Thanks 😄

**Pull request merged**